src/utils/calendar-time.js

```diff
@@ -0,0 +1,135 @@
+// Difference of this file to utils.js is that this file
```

Comment:
why the name `calendar-time`?

Reply:
Those are utilities for a calendar which respects time. It was really hard to find good names here. We need the utils to:
- share logic between the calendars (`DateInput` &  `DateRangInput`). `DateTimeInput` uses its own utils (this file).
- keep the date library (`moment`) out of our components which makes them more readable and makes it easier to replace the date-library if we ever want to